src/core/exec-invoke.c

```diff
@@ -4904,7 +4906,13 @@ int exec_invoke(
                         u = NULL;
 
                 if (u) {
-                        r = get_fixed_user(u, &username, &uid, &gid, &pwent_home, &shell);
+                        /* We can't use nss unconditionally for root without risking deadlocks if some IPC services
+                         * will be started by pid1 and are ordered after us. But if SetLoginEnvironment= is
+                         * enabled explicitly, or PAM shall be invoked, let's consult NSS even for root,
```

Comment:
Are there any implicit ordering dependency or so to avoid deadlocks when these settings specified??